File: mifosng-android/src/main/java/com/mifos/services/data/CenterPayload.java
/*
 * This project is licensed under the open source MPL V2.
 * See https://github.com/openMF/android-client/blob/master/LICENSE.md
 */
package com.mifos.services.data;

import android.os.Parcel;
import android.os.Parcelable;

import com.mifos.api.local.MifosBaseModel;
import com.mifos.api.local.MifosDatabase;
import com.raizlabs.android.dbflow.annotation.Column;
import com.raizlabs.android.dbflow.annotation.ModelContainer;
import com.raizlabs.android.dbflow.annotation.PrimaryKey;
import com.raizlabs.android.dbflow.annotation.Table;

/**
 * Created by nellyk on 1/22/2016.
 */
@Table(database = MifosDatabase.class)
@ModelContainer
public class CenterPayload extends MifosBaseModel implements Parcelable {

    @PrimaryKey(autoincrement = true)
    transient int id;

    @Column
    transient String errorMessage;

    @Column
    String dateFormat;

    @Column
    String locale;

    @Column
    private String name;

    @Column
    private int officeId;

    @Column
    private boolean active;

    @Column
    private String activationDate;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getErrorMessage() {
        return errorMessage;
    }

    public void setErrorMessage(String errorMessage) {
        this.errorMessage = errorMessage;
    }

    public String getDateFormat() {
        return dateFormat;
    }

    public void setDateFormat(String dateFormat) {
        this.dateFormat = dateFormat;
    }

    public String getLocale() {
        return locale;
    }

    public void setLocale(String locale) {
        this.locale = locale;
    }


    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getOfficeId() {
        return officeId;
    }

    public void setOfficeId(int officeId) {
        this.officeId = officeId;
    }

    public boolean isActive() {
        return active;
    }

    public void setActive(boolean active) {
        this.active = active;
    }

    public String getActivationDate() {
        return activationDate;
    }

    public void setActivationDate(String activationDate) {
        this.activationDate = activationDate;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.dateFormat);
        dest.writeString(this.locale);
        dest.writeString(this.name);
        dest.writeValue(this.officeId);
        dest.writeValue(this.active);
        dest.writeString(this.activationDate);
    }

    public CenterPayload() {
    }

    protected CenterPayload(Parcel in) {
        this.dateFormat = in.readString();
        this.locale = in.readString();
        this.name = in.readString();
        this.officeId = (Integer) in.readValue(Integer.class.getClassLoader());
        this.active = (Boolean) in.readValue(Boolean.class.getClassLoader());
        this.activationDate = in.readString();
    }

    public static final Parcelable.Creator<CenterPayload> CREATOR =
            new Parcelable.Creator<CenterPayload>() {
                @Override
                public CenterPayload createFromParcel(Parcel source) {
                    return new CenterPayload(source);
                }

                @Override
                public CenterPayload[] newArray(int size) {
                    return new CenterPayload[size];
                }
            };


}

File: mifosng-android/src/main/java/com/mifos/services/data/CenterPayload.kt
/*
 * This project is licensed under the open source MPL V2.
 * See https://github.com/openMF/android-client/blob/master/LICENSE.md
 */
package com.mifos.services.data

import android.os.Parcel
import android.os.Parcelable
import android.os.Parcelable.Creator
import com.mifos.api.local.MifosBaseModel
import com.mifos.api.local.MifosDatabase
import com.raizlabs.android.dbflow.annotation.Column
import com.raizlabs.android.dbflow.annotation.ModelContainer
import com.raizlabs.android.dbflow.annotation.PrimaryKey
import com.raizlabs.android.dbflow.annotation.Table

/**
 * Created by nellyk on 1/22/2016.
 */
@Table(database = MifosDatabase::class)
@ModelContainer
class CenterPayload : MifosBaseModel, Parcelable {
    @JvmField
    @PrimaryKey(autoincrement = true)
    @Transient
    var id = 0

    @JvmField
    @Column
    @Transient
    var errorMessage: String? = null

    @JvmField
    @Column
    var dateFormat: String? = null

    @JvmField
    @Column
    var locale: String? = null

    @Column
    var name: String? = null

    @Column
    var officeId = 0

    @Column
    var isActive = false

    @Column
    var activationDate: String? = null
    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(dateFormat)
        dest.writeString(locale)
        dest.writeString(name)
        dest.writeValue(officeId)
        dest.writeValue(isActive)
        dest.writeString(activationDate)
    }

    constructor() {}
    protected constructor(`in`: Parcel) {
        dateFormat = `in`.readString()
        locale = `in`.readString()
        name = `in`.readString()
        officeId = (`in`.readValue(Int::class.java.classLoader) as Int?)!!
        isActive = (`in`.readValue(Boolean::class.java.classLoader) as Boolean?)!!
        activationDate = `in`.readString()
    }

    companion object {
        @JvmField
        val CREATOR: Creator<CenterPayload> = object : Creator<CenterPayload> {
            override fun createFromParcel(source: Parcel): CenterPayload? {
                return CenterPayload(source)
            }

            override fun newArray(size: Int): Array<CenterPayload?> {
                return arrayOfNulls(size)
            }
        }
    }
}